public/pages/Alerts/components/CorrelationAlertFlyout/CorrelationAlertFlyout.tsx

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+    EuiBasicTable,
+    EuiBasicTableColumn,
+    EuiButton,
+    EuiButtonIcon,
+    EuiFlexGroup,
+    EuiFlexItem,
+    EuiFlyout,
+    EuiFlyoutBody,
+    EuiFlyoutHeader,
+    EuiLink,
+    EuiSpacer,
+    EuiTitle,
+  } from '@elastic/eui';
+  import { RuleSource } from '../../../../../server/models/interfaces';
+  import React from 'react';
+  import { ContentPanel } from '../../../../components/ContentPanel';
+  import { ALERT_STATE, DEFAULT_EMPTY_DATA, ROUTES } from '../../../../utils/constants';
+  import {
+    capitalizeFirstLetter,
+    createTextDetailsGroup,
+    errorNotificationToast,
+    formatRuleType,
+    renderTime,
+  } from '../../../../utils/helpers';
+  import { IndexPatternsService, OpenSearchService, DetectorsService } from '../../../../services';
+  import { parseAlertSeverityToOption } from '../../../CreateDetector/components/ConfigureAlerts/utils/helpers';
+  import { NotificationsStart } from 'opensearch-dashboards/public';
+  import { DataStore } from '../../../../store/DataStore';
+  import { CorrelationAlertTableItem, Finding } from '../../../../../types';
+  
+  export interface CorrelationAlertFlyoutProps {
+    alertItem: CorrelationAlertTableItem;
+    notifications: NotificationsStart;
+    onClose: () => void;
+    onAcknowledge: (selectedItems: CorrelationAlertTableItem[]) => void;
+  }
+  
+  export interface CorrelationAlertFlyoutState {
+    acknowledged: boolean;
+    findingItems: Finding[];
+    loading: boolean;
+    rules: { [key: string]: RuleSource };
+  }
+  
+  export class CorrelationAlertFlyout extends React.Component<CorrelationAlertFlyoutProps, CorrelationAlertFlyoutState> {
+    constructor(props: CorrelationAlertFlyoutProps) {
+      super(props);
+  
+      this.state = {
+        acknowledged: props.alertItem.state === ALERT_STATE.ACKNOWLEDGED,
+        findingItems: [],
+        loading: false,
+        rules: {},
+      };
+    }
+  
+    async componentDidMount() {
+      this.getFindings();
+    }
+  
+    getFindings = async () => {
```

Comment:
Dedupe these utility methods between the regular alert flyout and this one

Reply:
will raise a seperate PR to refactor this.